lib/components/base-components/Renderable.ts

```diff
@@ -179,11 +179,27 @@ export abstract class Renderable implements IRenderable {
     // Skip if component is being removed and not initialized
     if (!isInitialized && this.shouldBeRemoved) return
 
+    const emitRenderEvent = (eventType: "start" | "end") => {
+      const eventPayload = { renderId: this._renderId }
+      const eventName = `renderable:renderLifecycle:${phase}:${eventType}`
+      ;(this as any).root
+        ?.emit(
+          eventName,
+          eventPayload,
+        )(this as any)
+        .root?.emit("renderable:renderLifecycle:anyEvent", {
+          ...eventPayload,
+          eventName,
+        })
+    }
```

Comment:
don't use an inline function here- make it a proper class method `this._emitRenderLifecycleEvent(renderPhase, "start")`